persistence/pre_persistence/persistence.go

```diff
@@ -0,0 +1,306 @@
+package pre_persistence
+
+import (
+	"encoding/hex"
+	"fmt"
+	"github.com/pokt-network/pocket/shared/config"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/pokt-network/pocket/shared/types"
+	"strings"
+
+	"github.com/jordanorelli/lexnum"
+	"github.com/syndtr/goleveldb/leveldb/comparer"
+	"github.com/syndtr/goleveldb/leveldb/memdb"
+	"github.com/syndtr/goleveldb/leveldb/util"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	FirstSavePointKey                                        = []byte("first_savepoint_key")
+	DeletedPrefixKey                                         = []byte("deleted/")
+	BlockPrefix                                              = []byte("block/")
+	TransactionKeyPrefix                                     = []byte("transaction/")
+	PoolPrefixKey                                            = []byte("pool/")
+	AccountPrefixKey                                         = []byte("account/")
+	AppPrefixKey                                             = []byte("app/")
+	UnstakingAppPrefixKey                                    = []byte("unstaking_app/")
+	ServiceNodePrefixKey                                     = []byte("service_node/")
+	UnstakingServiceNodePrefixKey                            = []byte("unstaking_service_node/")
+	FishermanPrefixKey                                       = []byte("fisherman/")
+	UnstakingFishermanPrefixKey                              = []byte("unstaking_fisherman/")
+	ValidatorPrefixKey                                       = []byte("validator/")
+	UnstakingValidatorPrefixKey                              = []byte("unstaking_validator/")
+	ParamsPrefixKey                                          = []byte("params/")
+	_                             modules.PersistenceModule  = &PrePersistenceModule{}
+	_                             modules.PersistenceContext = &PrePersistenceContext{}
+)
+
+type PrePersistenceModule struct {
+	CommitDB *memdb.DB
+	Mempool  types.Mempool
+	//pocketBusMod modules.Bus  TODO: add back in once bus is integrated
+	Cfg *config.Config
+}
+
+func NewPrePersistenceModule(commitDB *memdb.DB, mempool types.Mempool, cfg *config.Config) *PrePersistenceModule {
+	return &PrePersistenceModule{CommitDB: commitDB, Mempool: mempool, Cfg: cfg}
+}
+
+func (m *PrePersistenceModule) NewContext(height int64) (modules.PersistenceContext, error) {
+	newDB := NewMemDB()
+	it := m.CommitDB.NewIterator(&util.Range{Start: HeightKey(height, nil), Limit: HeightKey(height+1, nil)})
```

Comment:
(_minor comment_) I think for long pieces of code like this, it's very readable to put it on multiple lines:
```
it := m.CommitDb.NewIterator(&util.Range{
    Start: HeightKey(height, nil),
    Limint: HeightKey(height_1, nil),
})
```
and only fully use the for semantics:
```
defer it.Release()
for it.First(); it.Valid(); it.Next() {
  // ...
}
```